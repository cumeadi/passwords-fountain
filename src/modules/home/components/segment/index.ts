export { Segment } from './segment.component';
