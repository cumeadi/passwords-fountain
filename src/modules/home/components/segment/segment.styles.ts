import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { theme, typography, media, stylelint, negative } from '@/common/theme';
setPragma(h);

export const Wrapper = styled('section')``;

export const TitleBar = styled('div')`
    width: calc(100% + (2 * ${theme.spacing.m18}));
    margin-left: ${negative(theme.spacing.m18)};
    padding: ${theme.spacing.s12} ${theme.spacing.m18};
    background: ${theme.colors.blue};

    ${media.gte(theme.breakpoints.m880)(stylelint.css`
        width: calc(100% + ${theme.spacing.xxl60});
        margin-left: ${negative(theme.spacing.xl30)};
        padding: ${theme.spacing.m18} ${theme.spacing.xxl60};
    `)}
`;

export const Title = styled('h3')`
    color: ${theme.colors.white};
    font-family: ${theme.fontFamilies.bold};
    ${typography.text20}

    ${media.gte(theme.breakpoints.s480)(stylelint.css`
        ${typography.text24}
    `)}
`;

export const ContentWrapper = styled('div')`
    padding: ${theme.spacing.m18} 0;
    ${typography.text18}

    ${media.gte(theme.breakpoints.m880)(stylelint.css`
        padding: ${theme.spacing.xl40} ${theme.spacing.xl30};
        ${typography.text24}
    `)}
`;
