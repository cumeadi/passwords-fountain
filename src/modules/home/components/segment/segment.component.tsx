import { h, VNode } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import PropTypes from 'prop-types';
import { Wrapper, TitleBar, Title, ContentWrapper } from './segment.styles';
import { Text } from '@/modules/localisation/localisation.context';

export const Segment: TypedComponent<Props> = ({
    id,
    title,
    children,
}: Props) => {
    const rootProps = id ? { id } : {};

    return (
        <Wrapper {...rootProps}>
            <TitleBar>
                <Title>
                    <Text>{title}</Text>
                </Title>
            </TitleBar>
            <ContentWrapper>{children}</ContentWrapper>
        </Wrapper>
    );
};

interface Props {
    id?: string;
    title: string;
    children: VNode | VNode[];
}

Segment.propTypes = {
    id: PropTypes.string,
    title: PropTypes.string.isRequired,
    children: PropTypes.any.isRequired,
};
