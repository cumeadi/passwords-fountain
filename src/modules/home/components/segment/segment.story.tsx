import { h, VNode } from 'preact';
import { Segment } from './segment.component';
import { withA11y } from '@storybook/addon-a11y';

export default {
    title: 'Segment',
    decorators: [withA11y],
};

export const defaultView = (): VNode => (
    <Segment title="Title test">
        <div>Test content</div>
    </Segment>
);
