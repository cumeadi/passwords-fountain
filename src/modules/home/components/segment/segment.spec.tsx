import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { Segment } from './segment.component';

describe('Segment', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<Segment />);

        expect(asFragment()).toMatchSnapshot();
    });
});
