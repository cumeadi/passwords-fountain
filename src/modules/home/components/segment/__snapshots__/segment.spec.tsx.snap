// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segment should render correctly 1`] = `
<DocumentFragment>
  <section
    class="go115679"
  >
    <div
      class="go4161270645"
    >
      <h3
        class="go1148060721"
      />
    </div>
    <div
      class="go3253515101"
    />
  </section>
</DocumentFragment>
`;
